### PR TITLE
Add linefeed in the mdx and handle it

### PR DESCRIPTION
**Pull request by harish-sethuraman**
Problem:
Multiple commands are displayed in the same line with line feed missing. 
page: https://beta.reactjs.org/learn/start-a-new-react-project
Solution: 
- Add preserve whitespace to classes
- ~~We can use 2 trailing empty spaces or `<br/>` to add new line feed~~
- ~~when we have multiple lines `typeof children.props.children` is not string but an object. So use children.props.children directly when it is a string or an object (`typeof children.props.children === 'string'` )~~

Before:
Screenshot 2021-11-04 at 11 20 52 PM
After:
Screenshot 2021-11-04 at 11 21 04 PM

### CODE REVIEW

**Comment by gaearon** [comment]
>We can use 2 trailing empty spaces or <br/> to add new line feed

This doesn't make sense to me.

Can you make it work normally?

File: beta/src/components/MDX/TerminalBlock.tsx [hunk 1]
```diff
@@ -28,8 +28,7 @@ function TerminalBlock({level = 'info', children}: TerminalBlockProps) {
   if (typeof children === 'string') {
     message = children;
   } else if (
-    React.isValidElement(children) &&
-    typeof children.props.children === 'string'
+    React.isValidElement(children)
```
Comment (gaearon) [inline beta/src/components/MDX/TerminalBlock.tsx:31]: Do we understand why this transformation exists at all? What happens if you remove it altogether?

**Comment by harish-sethuraman** [comment]
> > We can use 2 trailing empty spaces or  to add new line feed
> 
> This doesn't make sense to me.
> 
> Can you make it work normally?

we could make it work normally but we need to render the children on the whole instead of doing `children.props.children`. Since the children.props.children is considered as a string and not a react element new line is not added 🙁 . 

I was trying to look around to add empty spaces in markdowns and found this 

https://www.markdownguide.org/basic-syntax/

Screenshot 2021-11-05 at 12 13 02 AM

File: beta/src/components/MDX/TerminalBlock.tsx (hunk 1 above)
Comment (harish-sethuraman) [reply beta/src/components/MDX/TerminalBlock.tsx:31]: I hope the condition was added to render the strings directly and render the children if it has collection of elements together (Like the commands in two lines).

This check can be removed and we can render the children of terminalBlocks directly too. But when we render the children directly the texts are wrapped inside `<p>` and takes extra space like this.
Screenshot 2021-11-05 at 12 27 54 AM

**Comment by gaearon** [comment]
How does this work inside triple back tick (code blocks)? Terminal blocks should be treated similarly.

**Comment by gaearon** [comment]
Does setting `whiteSpace: 'pre'` in the style fix it?

**Comment by harish-sethuraman** [comment]
Adding whitespace-pre fixed the issue 👀 . I've added that to the terminal block now. Thanks.

**Comment by gaearon** [comment]
Thx!

**Pull request merged**